### PR TITLE
Extract BlockLoader from execution Ingestion engine

### PR DESCRIPTION
**Pull request by zhangchiqing**
Review https://github.com/onflow/flow-go/pull/4713 first

This PR extracts the BlockLoader logic from ingestion engine in order to reduce the complexity of the ingestion engine.

The BlockLoader is responsible for loading blocks to be re-executed on startup.

### CODE REVIEW

File: engine/execution/ingestion/engine_test.go [hunk 1]
```diff
@@ -1546,264 +1550,13 @@ func newIngestionEngine(t *testing.T, ps *mocks.ProtocolState, es *mockExecution
...
 			false,
 		),
+		loader,
 	)
 
 	require.NoError(t, err)
 	return engine, headers
 }
 
-func logChain(chain []*flow.Block) {
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine_test.go:1555]: all moved to loader_test

File: engine/execution/ingestion/engine.go [hunk 2]
```diff
@@ -310,68 +200,6 @@ func (e *Engine) reloadUnexecutedBlocks() error {
 		blockByCollection *stdmap.BlockByCollectionBackdata,
 		executionQueues *stdmap.QueuesBackdata,
 	) error {
-
-		// saving an executed block is currently not transactional, so it's possible
```
Comment (peterargue) [inline engine/execution/ingestion/engine.go:314]: this section of code was removed. is it not needed anymore?

Comment (zhangchiqing) [reply engine/execution/ingestion/engine.go:314]: Good catch. 

Actually, the code is in `BlockLoader.LoadUnexecuted` method, I just need to copy the comments over

File: engine/execution/ingestion/engine.go [hunk 3]
```diff
@@ -179,126 +182,13 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 	return nil
 }
 
-func (e *Engine) finalizedUnexecutedBlocks(finalized protocol.Snapshot) (
```
Comment (zhangchiqing) [inline engine/execution/ingestion/engine.go:182]: all moved to loader

**Review by peterargue — APPROVED** [review]

**Review by ramtinms — APPROVED** [review]

**Pull request merged**